weather_providers/smhi.py

```diff
@@ -0,0 +1,138 @@
+                        15: "Light snow showers",
+                        16: "Moderate snow showers",
+                        17: "Heavy snow showers",
+                        18: "Light rain",
+                        19: "Moderate rain",
+                        20: "Heavy rain",
+                        21: "Thunder",
+                        22: "Light sleet",
+                        23: "Moderate sleet",
+                        24: "Heavy sleet",
+                        25: "Light snowfall",
+                        26: "Moderate snowfall",
+                        27: "Heavy snowfall",
+                    }
+        description = description_dict[weathercode]
+
+        logging.debug(
+            "get_description_by_weathercode({}) - {}"
+            .format(weathercode, description))
+
+        return description.title()
+
+    # Get weather from SMHI API
+    # https://opendata.smhi.se/apidocs/metfcst/get-forecast.html#get-point-forecast
+    # The API response is a complete forecast approximately 10 days ahead of the latest current forecast.
+    # All times in the answer given in UTC.
+    # Precipitation parameters have a distribution in time (a time interval) until the valid time for current data.
+    # The interval starts at the time step before. At the beginning of the forecast, the interval is one hour.
+    # Later in the forecast, the time interval increases (eg 3, 6 and 12 h). Unit remains mm / h.
+    # So, current hour is index 0, next hour is index 1
+    # The API accepts 6 decimals in the lon/lat. With more decimals, it returns 404.
+
+    def get_weather(self):
+
+        url = ("https://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/{}/lat/{}/data.json"
+               .format(self.location_long, self.location_lat))
+
+        headers = {"User-Agent": self.smhi_self_id}
+
+        response_data = self.get_response_json(url, headers=headers)
+        logging.debug(response_data)
+        weather_data = response_data["timeSeries"]
+        logging.debug("get_weather() - {}".format(weather_data))
+
+        # Get the weather code of the first item.
+        for data in weather_data[0]["parameters"]:
+            if data["name"] == "Wsymb2":
+                weather_code = data["values"][0]
+
+        daytime = self.is_daytime(self.location_lat, self.location_long)
+
+        # { "temperatureMin": "2.0", "temperatureMax": "15.1", "icon": "mostly_cloudy", "description": "Cloudy with light breezes" }
+        # No Min or Max here. We just get the estimated temperature for the hour.
+        # Since we get the forecast for several hours and days, get the min/max for the next 12 hours?
+        weather = {}
+        temp_list = []
+
+        for item in range(0, 12):
+            for param in weather_data[item]['parameters']:
+                if param['name'] == 't':
+                    temp = param['values'][0]
+                    temp_list.append(temp)
```

Comment:
Looping through 12 forecasts, getting the temperature out, adding it to a list, then below getting min and max of those temperatures. So it acts as a min/max forecast for 12 hours.  

The weather symbol unfortunately that's a bit difficult since there's no "min/max" or "average" of symbols.